cmd/testrunner/main.go

```diff
@@ -201,6 +208,30 @@ func withTestImage(image string) func(obj runtime.Object) {
 	}
 }
 
+// withPreHookImage sets the value of the PRE_STOP_HOOK_IMAGE
+// EnvVar from first container to `image`. The EnvVar is updated
+// if it exists. Or appended if there is no EnvVar with this `Name`.
+func withPreHookImage(image string) func(runtime.Object) {
+	return func(obj runtime.Object) {
+		if dep, ok := obj.(*appsv1.Deployment); ok {
+			preHookEnv := corev1.EnvVar{
+				Name:  "PRE_STOP_HOOK_IMAGE",
+				Value: image,
+			}
+			found := false
+			for idx := range dep.Spec.Template.Spec.Containers[0].Env {
+				if dep.Spec.Template.Spec.Containers[0].Env[idx].Name == preHookEnv.Name {
+					dep.Spec.Template.Spec.Containers[0].Env[idx].Value = preHookEnv.Value
+					found = true
+				}
+			}
+			if !found {
+				dep.Spec.Template.Spec.Containers[0].Env = append(dep.Spec.Template.Spec.Containers[0].Env, preHookEnv)
+			}
```

Comment:
I don't think it should ever not be present. This reads from `operator.yaml` which will always have it configured. We should just only ever need to replace it.